tbxforms/fields.py

```diff
@@ -159,6 +204,20 @@ def clean(self, value):
                 field.widget.errors.extend(
                     m for m in e.messages if m not in field.widget.errors
                 )
+
+        # check if the given month and day are valid for the given year, this
+        # relies on the fields being declared in the order day, month, year
+        if len(clean_data) == 3:
+            day, month, year = map(int, clean_data)
+            dim = _days_in_month(year, month)
+            if day > dim:
+                error = _(
+                    "Enter a day between 1 and %(dim)s for the month and year "
+                    "you entered"
```

Comment:
**Comment**: I think this message is clear, though maybe `from 1 to %(dim)s` would be more literally correct. "Between 1 and 30" also seems fine colloquially.

Note that the [Error Messages section of Date Input | GOV.UK Design System](https://design-system.service.gov.uk/components/date-input/#error-messages) has a lot to say on this matter and is worth a read.

Reply:
I think this is now a typo in db65d80: "between 1 to %(dim)s".

Reply:
Fixed in [`449f085`](https://github.com/torchbox/tbxforms/pull/29/commits/449f0853079d1aaea40fb41aff78bfdeffff4033)